mtags/src/main/scala/scala/meta/internal/mtags/SymbolIndexBucket.scala

```diff
@@ -236,8 +264,13 @@ class SymbolIndexBucket(
     docs.documents.foreach { document =>
       document.occurrences.foreach { occ =>
         if (occ.symbol.isGlobal && occ.role.isDefinition) {
-          val acc = definitions.getOrElse(occ.symbol, Set.empty)
-          definitions.put(occ.symbol, acc + SymbolLocation(file, occ.range))
+          val toAdd = SymbolLocation(file, occ.range)
+          definitions.updateWith(occ.symbol) {
+            case Some(acc) => Some(acc + toAdd)
+            case None => Some(Set(toAdd))
+          }
```

Comment:
I have a dilemma. SymbolIndexBucket uses TrieMap to store toplevels and definitns, but those maps aren't updated in atomic way, `get` & then `set` can override some change in between. On the other hand, definitions seems to be updated sequentially. What do you think?

Reply:
I was thinking to replace some TrieMap usage by ConcurrentHashMap because of that but haven't faced a real incident

Reply:
Problem, if any, is very rare, it's the same case where `Set`s are needed, but more rare. It can only happen when there are more than one global symbols and they happened to be indexed at the same time. 
Definitions should be relatively safe, because they're updated in lazy way, as a consequence of calling `query0` method and this methos is sequential. On the other hand, there's no any limitation on calling `query` on parallel. Even though I also didn't observe any issue, I'd lean toward writing code which prevents such scenarios.

Reply:
I didn't mean it didn't make much sense. Only that I didn't find the time + enough motivation to do it.

I think it would be better to replace to avoid any even if it's rare issue.

Reply:
issue created https://github.com/scalameta/metals/issues/5072